File: 99.example/3.knight/cpp/knight.cpp
#include <iostream>
#include <string>

using namespace std;

const int CASE_SIZE=8; 
const int STEP_SIZE=3;

int main() {
    int ret = 0;
    string pos;
    getline(cin, pos);

    string arr[CASE_SIZE] = {"UUL", "UUR", "DDL", "DDR", "LLU", "LLD", "RRU", "RRD"};

    int row = pos[1] - '0';   
    int col = pos[0] - 'a' + 1;

    for(int i=0; i<CASE_SIZE; i++) {        
        for(int j=0; j<STEP_SIZE; j++) {
            if(arr[i][j] == 'U') {
                row--;
            } else if(arr[i][j] == 'D') {
                row++;
            } else if(arr[i][j] == 'L') {
                col--;
            } else if(arr[i][j] == 'R') {
                col++;
            }
        }
        
        if( (row <= 8) && (row >= 1) && (col <= 8) && (col >= 1) ) {                            
            ret++;
        }
        
        row = pos[1] - '0'; 
        col = pos[0] - 'a' + 1;
             
    }

    cout << ret << endl;
    return 0;

}

File: 99.example/3.knight/python/knight.py
data = input()

row = int(data[1])
col = int(ord(data[0])) - int(ord('a')) + 1

ss = [(-2,-1), (-1,-2), (1,-2), (2, -1), (2,1), (1,2), (-1,2), (-2,1)]

ret = 0

for s in ss:
    n_row = row + s[0]
    n_col = col + s[1]

    if n_row>=1 and n_row<=8 and n_col >=1 and n_col <=8:
        ret +=1

print(ret)
